dbms/src/Interpreters/SyntaxAnalyzer.cpp

```diff
@@ -49,29 +49,30 @@ using LogAST = DebugASTLog<false>; /// set to true to enable logs
 using Aliases = std::unordered_map<String, ASTPtr>;
 
 /// Add columns from storage to source_columns list.
-NamesAndTypesList collectSourceColumns(NamesAndTypesList source_columns, ASTSelectQuery * select_query,
-                                       const Context & context, StoragePtr & storage);
+void collectSourceColumns(NamesAndTypesList & source_columns, ASTSelectQuery * select_query,
```

Comment:
Do we change source_columns and select_query inside? 
It's a best practice to order unchanged parameter first and results next to them.